### PR TITLE
Electron-445 (Added a new API to keep electron in sync with client app user meeting status)

### PR DESCRIPTION
**Pull request by KiranNiranjan**
## Description
Added a new API to keep electron in sync with client app user presence status [ELECTRON-445](https://perzoinc.atlassian.net/browse/ELECTRON-445)

## Approach
How does this change address the problem?
#### Problem with the code:
 - The application was auto reloaded even when the user was in a meeting
#### Fix:
 - Added an API that keeps the Electron in sync with the client app user presence status
 - If the user presence status is `AWAY` the application will not be refreshed

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------
SFE-Client-App | [link](https://github.com/SymphonyOSF/SFE-Client-App/pull/9634)
SFE-Client-App(r52) | [link](https://github.com/SymphonyOSF/SFE-Client-App/pull/9635)

## Open Questions if any and Todos
- [X] Unit-Tests
- [X] Documentation
- [X] Automation-Tests
When solved, check the box and explain the answer.

### CODE REVIEW

File: js/memoryMonitor.js [hunk 1]
```diff
@@ -47,14 +48,35 @@ function optimizeMemory(memoryInfo) {
...
     }
 }
 
+/**
+ * Checks the user presence status to see
+ * if the user is active
+ * @return {boolean}
+ */
+function isUserActive() {
+    return !(userPresenceStatus && userPresenceStatus === 'AWAY');
```
Comment (VishwasShashidhar) [inline js/memoryMonitor.js]: can we abstract `AWAY` to a constant?

File: js/preload/preloadMain.js [hunk 2]
```diff
@@ -64,6 +64,13 @@ const throttledSetBadgeCount = throttle(1000, function(count) {
     });
 });
 
+const throttledSetPresenceStatus = throttle(1000, function(status) {
```
Comment (VishwasShashidhar) [inline js/preload/preloadMain.js]: so, this is every second right? if so, can we look at increasing the period to 10 seconds?

**Comment by VikasShashidhar** [comment]
@KiranNiranjan I am not sure if this approach is the right one. What if the user manually sets his/her presence status to "Away"? We can't really rely on this approach then.

I agree with @lneir. We'll need to rely on WebRTC to tell us if the user is active.

**Comment by KiranNiranjan** [comment]
@lneir @VikasShashidhar @VishwasShashidhar Updated the logic. Please review. Thanks 😃

File: js/memoryMonitor.js [hunk 3]
```diff
@@ -33,26 +35,47 @@ function gatherMemory() {
...
+ * @param cpuUsage
  */
-function optimizeMemory(memoryInfo) {
+function optimizeMemory(memoryInfo, cpuUsage) {
     const memoryConsumed = (memoryInfo && memoryInfo.workingSetSize / 1024) || 0;
     const canReload = (!reloadedTimeStamp || (new Date().getTime() - reloadedTimeStamp) > reloadThreshold);
 
-    if (memoryConsumed > maxMemory && systemIdleTime.getIdleTime() > maxIdleTime && canReload) {
+    if (memoryConsumed > maxMemory
+        && systemIdleTime.getIdleTime() > maxIdleTime
```
Comment (VikasShashidhar) [inline js/memoryMonitor.js]: Fix linting please

File: js/memoryMonitor.js [hunk 4]
```diff
@@ -33,26 +35,47 @@ function gatherMemory() {
...
             });
     }
 }
 
+/**
+ * Sets the current user meeting status
+ * @param bool - Whether user is in an active meeting
+ */
+function setIsInMeeting(bool) {
+    isInMeeting = bool;
```
Comment (VikasShashidhar) [inline js/memoryMonitor.js]: Please rename var

File: js/memoryMonitor.js (hunk 3 above)
Comment (KiranNiranjan) [reply js/memoryMonitor.js]: I don't see any linting issue here. Can you please let me know what is missing?

**Pull request merged**